new changes

Tejaswini3107 opened the pull request.

Tejaswini3107 committed:
new changes

UmarMahmoodShaikh commented on lib/gotham/accounts/role.ex:
```diff
@@ -15,11 +15,11 @@ defmodule Gotham.Accounts.Role do
 
-  # Numeric role IDs (enum mapping)
-  @admin_id 1
+  # Numeric role IDs (enum mapping) - updated to match actual database IDs
```
Please revert back these changes. 
We need Admin Id to be 1
Employee to be 4

Tejaswini3107 committed:
feat: Complete 87-endpoint Time Manager API with comprehensive role-based access control

- Implement full authentication system with dual endpoint formats (sign_up/sign_in + register/auth_login)
- Add comprehensive 87-endpoint API across 9 categories:
  * Authentication & User Management
  * Time Tracking & Clock Management
  * Break Management & GPS Tracking
  * Analytics & Reporting
  * Payroll & Notifications
  * Settings & Approvals
- Establish role-based access control for Employee/Manager/Admin users
- Fix role ID mappings: Admin=1, Manager=2, HR=3, Employee=4
- Add CORS support and content negotiation
- Implement JWT-based authentication with proper token validation
- Create comprehensive controller structure with proper authorization checks
- Validate role-based permissions across all endpoint categories
- Ensure data isolation and security boundaries between user types
